winpr/libwinpr/utils/wlog/UdpAppender.c

```diff
@@ -0,0 +1,209 @@
+/**
+ * WinPR: Windows Portable Runtime
+ * WinPR Logger
+ *
+ * Copyright © 2015 Thincast Technologies GmbH
+ * Copyright © 2015 David FORT <contact@hardening-consulting.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+
+#include <winpr/crt.h>
+#include <winpr/environment.h>
+#include <winpr/thread.h>
+
+#include <winpr/wlog.h>
+
+#include "wlog/Message.h"
+#include "wlog/UdpAppender.h"
+
+#ifndef _WIN32
+#include <unistd.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#endif
+
+static BOOL WLog_UdpAppender_Open(wLog* log, wLogUdpAppender* appender)
+{
+	char addressString[256];
+	struct addrinfo hints;
+	struct addrinfo* result;
+	int status, addrLen;
+	char *colonPos;
+
+
+	if (!log || !appender)
+		return FALSE;
+
+	if (appender->targetAddrLen) /* already opened */
+		return TRUE;
+
+	colonPos = strchr(appender->host, ':');
+	if (!colonPos)
+		return FALSE;
+	addrLen = colonPos - appender->host;
+	memcpy(addressString, appender->host, addrLen);
+	addressString[addrLen] = '\0';
+
+	ZeroMemory(&hints, sizeof(hints));
+	hints.ai_family = AF_INET;
+	hints.ai_socktype = SOCK_DGRAM;
+
+	status = getaddrinfo(addressString, colonPos+1, &hints, &result);
+	if (status != 0)
+		return FALSE;
+
+	if (result->ai_addrlen > sizeof(appender->targetAddr))
+	{
+		freeaddrinfo(result);
+		return FALSE;
+	}
+
+	memcpy(&appender->targetAddr, result->ai_addr, result->ai_addrlen);
+	appender->targetAddrLen = result->ai_addrlen;
+
+
+	return TRUE;
+}
+
+BOOL Wlog_UdpAppender_SetTarget(wLogUdpAppender* appender, const char *host)
+{
+
+	appender->targetAddrLen = 0;
+	if (appender->host)
+		free(appender->host);
+
+	appender->host = _strdup(host);
+	return (appender->host != NULL) && WLog_UdpAppender_Open(NULL, appender);
```

Comment:
I guess WLog_UdpAppender_Open() would always return FALSE here because NULL is passed for the log parameter which triggers the return FALSE in line 50